test/Test/Password.hs

```diff
@@ -0,0 +1,21 @@
+module Test.Password
+       ( pwdHashVerify
+       ) where
+
+import Hedgehog (MonadGen, Property, assert, forAll, property)
+
+import CakeSlayer.Password (PasswordPlainText (..), mkPasswordHashWithPolicy, verifyPassword)
+
+import qualified Crypto.BCrypt as BC
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+
+pwdHashVerify :: Property
+pwdHashVerify = property $ do
+    randomPwd <- forAll genPwd
+    let hashPwd = mkPasswordHashWithPolicy BC.fastBcryptHashingPolicy randomPwd
+    whenJustM hashPwd $ \pwdHash -> assert $ verifyPassword randomPwd pwdHash
```

Comment:
I just noticed that our test can succed if password generated `Nothing`. I guess for more robustness it's better to write tests like this (bonus points for removing `fromJust`, I'm not sure whether there's `failure` combinator in hedgehog or not)

```haskell
pwdHash <- mkPasswordHashWithPolicy BC.fastBcryptHashingPolicy randomPwd
assert $ pwdHash /= Nothing && verifyPassword randomPwd (fromJust pwdHash)
```